src/app/material-docs-app.ts

```diff
@@ -24,9 +24,13 @@ export class MaterialDocsApp {
         if (!isNavigationWithinComponentView(previousRoute, urlAfterRedirects)) {
           resetScrollPosition();
         }
-
         previousRoute = urlAfterRedirects;
         ga.locationChanged(urlAfterRedirects);
+
+        // set skip link
+        if(!router.url.endsWith('#main-content')) {
+          this.skipLinkTarget = router.url + "#main-content";
```

Comment:
I could see this breaking for a URL like `.../stuff#not-the-main-content`. I think you should just be able to set the `href` to `#main-content` and the rest of the URL is assumed

Reply:
setting `href` directly to `#main-content` will cause the page to redirect to the homepage. 
a similar issue is addressed in [doc-viewer.ts](https://github.com/angular/material.angular.io/blob/master/src/app/shared/doc-viewer/header-link.ts)